src/com/cdd/bao/validator/RemappingException.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * BioAssay Ontology Annotator Tools
+ *
+ * (c) 2014-2018 Collaborative Drug Discovery Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License 2.0
+ * as published by the Free Software Foundation:
+ *
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.en.html
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+package com.cdd.bao.validator;
+
+import java.util.*;
+
+/*
+	Throw this exception if we detect a cyclical relationship in the dependency
+	graph implied by the term-remappings.
+ */
+
+public class RemappingException extends Exception
```

Comment:
I doubt this needs its own exception, but if you think its warranted, then make it into an inner class. Top level exceptions are only OK when they are intended to be used in a fair number of places. If it's just one block of code that throws it, then put it as close to that as possible. Or in this case, it's probably better to use some other exception. If we're not going to write code that traps that class specifically and does something with it that's different to the other possible exceptions, then the payload is limited the message & position, which means you can pick some other exception to use.